server/utils/helpers.js

```diff
@@ -0,0 +1,95 @@
+import axios from 'axios'
+import querystring from 'querystring'
+import consts from '../utils/consts'
+import jwtDecode from 'jwt-decode'
+
+export const decode = token => {
+  return token ? jwtDecode(token) : null
+}
+
+/**
+ * Handle possibility where token endpoint, at exp returns seconds instead of µ seconds
+ */
+export const handleTokenExp = exp => {
+  let out = exp
+
+  const milliseconds = new Date().getTime()
+  // const millisecondsDigitCount = ((milliseconds).toString()).length
+  const seconds = Math.floor(milliseconds / 1000)
+  const secondsDigitCount = ((seconds).toString()).length
+
+  const isExpressedInSeconds = ((exp).toString()).length === secondsDigitCount
+  // const isExpressedInMilliSeconds = ((exp).toString()).length === millisecondsDigitCount
+
+  // If the exp is 25 hours or less, adjust the time to miliseconds
+  // Otherwise let's not touch it
+  if (isExpressedInSeconds) {
+    const durationInSeconds = Math.floor((exp - seconds))
+    const hours = Math.floor(durationInSeconds / 3600)
+    if (hours < 25) { // Make 25 configurable?
+      out *= 1000
+    }
+  }
+
+  return out
+}
+
+/**
+ * Make an async off-the-band POST request.
+ *
+ * Notice that LB_ADDR can be superseeded to your own backend
+ * instead of mocking (static) endpoint.
+ *
+ * Differeciation factor is when you use /hpi, Koa will take care of it
+ * and yours MUST therefore NOT start by /hpi, and Koa will be out of the way.
+ *
+ * All of this is done when you set your own LB_ADDR environment setup
+ * to point to your own API.
+ */
+export const createRequest = async (method, url, requestConfig) => {
+  const {
+    payload = null,
+    ...restOfRequestConfig
+  } = requestConfig
+  let requestConfigObj = {
+    timeout: consts.AXIOS_DEFAULT_TIMEOUT,
```

Comment:
Another pain point was that timeout was too short, and wasn’t the same in both client/server.
Backends doesn’t always respond fast. Most notably ones in production with loads of data and across various networks.